Functions/FirebaseFunctions/FUNSerializer.m

```diff
@@ -153,7 +153,10 @@ - (nullable id)decodeWrappedType:(NSDictionary *)wrapped error:(NSError **)error
     NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
     NSNumber *n = [formatter numberFromString:value];
     if (n == nil) {
-      *error = FUNInvalidNumberError(value, wrapped);
+      // Only assign the `error` parameter if it was passed in.
+      if (error != NULL) {
```

Comment:
nit: Defining function like [this](https://github.com/firebase/firebase-ios-sdk/blob/42118c9b1398af076b9a05557b4cb8bb21ddc70a/FirebaseInstallations/Source/Library/Errors/FIRInstallationsErrorUtil.m#L23) in `GoogleUtilities` and using it in client SDKs may save us a couple lines here and there. WDYT?

Reply:
I don't mind the idea but I'm not sure it's worth it for a few reasons:

- It's obvious what this does at the call site,
- We'll have to remember to use it, which I'm not convinced I'd be able to since it's a very quick `if` check,
- More classes imported, and potentially even a new dependency

I think if it was littered throughout the codebase it'd be worth it (like your example with FIS) but maybe SDKs can decide that for themselves.